src/include/platform/internal/GenericConfigurationManagerImpl.cpp

```diff
@@ -243,6 +243,26 @@ CHIP_ERROR GenericConfigurationManagerImpl<ImplClass>::StoreManufacturingDate(co
 template <class ImplClass>
 void GenericConfigurationManagerImpl<ImplClass>::InitiateFactoryReset()
 {
+#if CHIP_ENABLE_ROTATING_DEVICE_ID
+    // Increment life time counter to protect against long-term tracking of rotating device ID.
+    _IncrementLifetimeCounter();
```

Comment:
why do we need to call this when initiating factory reset? Shouldn't it be fine to just increment each time we start advertising?

Reply:
i guess it should be fine to depend on advertisement to increment the counter, the factoryReset change was in the original code.